web/modules/weather_data/src/Service/DailyForecastTrait.php

```diff
@@ -196,11 +197,50 @@ public function getDailyForecastFromGrid(
             $alerts,
         );
 
+        // Get a mapping of the starTime for each day,
+        // starting with the today period. These will
+        // be used for grouping precipitation totals
+        // for each day.
+        $periodStartTimes = [
+            DateTimeUtility::stringToDate(
+                $todayPeriods[0]->startTime,
+                $timezone,
+            ),
+        ];
+        $detailedPeriodStartTimes = array_map(function ($periodPair) use (
+            &$timezone,
+        ) {
+            return DateTimeUtility::stringToDate(
+                $periodPair[0]->startTime,
+                $timezone,
+            );
+        }, array_chunk($detailedPeriods, 2));
+        $periodStartTimes = array_merge(
+            $periodStartTimes,
+            $detailedPeriodStartTimes,
+        );
+
+        // Get raw precipitation periods data, then map and
+        // chunk into groups of periods based on each day's
+        // startTime
+        $precipPeriods = $this->getHourlyPrecipitation($wfo, $x, $y, $now);
+        $precipPeriods = array_map(function ($startTime) use (
+            &$precipPeriods,
+            &$timezone,
+        ) {
+            return $this->filterHourlyPrecipitationToDay(
+                $startTime,
+                $precipPeriods,
+                $timezone,
+            );
+        }, $periodStartTimes);
```

Comment:
This all seems like it should work! The data structure is a little confusing, but that's simply _true_, not an implementation issue.